src/components/modals/revision-list.jsx

```diff
@@ -0,0 +1,58 @@
+import PropTypes from "prop-types";
+import React from "react";
+import styled from "react-emotion";
+import format from "date-fns/format";
+
+import List from "@material-ui/core/List";
+import ListItem from "@material-ui/core/ListItem";
+import ListItemText from "@material-ui/core/ListItemText";
+
+const RevisionListContainer = styled("div")`
+  overflow: auto;
+`;
+
+export class RevisionList extends React.Component {
+  static propTypes = {
+    revisionList: PropTypes.arrayOf(
+      PropTypes.shape({
+        created: PropTypes.string.isRequired,
+        id: PropTypes.number.isRequired
+      })
+    ),
+    selectedRevisionId: PropTypes.number,
+    revisionClicked: PropTypes.func.isRequired
+  };
+
+  render() {
+    return (
+      <RevisionListContainer>
+        <List>
+          <ListItem
+            button
+            key="local-changes"
+            onClick={() => this.props.revisionClicked(undefined)}
+            selected={this.props.selectedRevisionId === undefined}
+          >
+            <ListItemText primary="Unsaved Changes" />
+          </ListItem>
+          {this.props.revisionList &&
+            this.props.revisionList.map(revision => (
+              <ListItem
+                button
+                key={`revision-${revision.id}`}
+                onClick={() => this.props.revisionClicked(revision.id)}
+                selected={this.props.selectedRevisionId === revision.id}
+              >
+                <ListItemText
+                  primary={format(
+                    new Date(revision.created),
+                    "MMM dd, uuuu HH:MM"
+                  )}
+                />
+              </ListItem>
+            ))}
+        </List>
+      </RevisionListContainer>
```

Comment:
it would be good practice to write some test for this. not a ton, but at least a couple to verify that that correct items are selected and that the correct number of items are present. console-pane.test.js might be a place to look.

i also think that this element should probably be connected directly to the store. none of the things that are passed down here as props from HistoryModalUnconnected actually need to be passed in at that point (that component just passes them straight through via it's mapStateToProps, which is a good sign that a child component should connect to the store). also, the onclick function used here can be a method of this component, since the selection of items is really a concern of the list, not the modal.

refactoring that stuff should slightly streamline the history modal and separate these concerns a bit better. it will also give you two smaller and simpler mapStateToProps functions to test

Reply:
Hmm, I don't agree with connecting this to the store, I think it's much easier to reason about this component as a display-only thing. I think it would be confusing to have three different connections to the redux store for what is essentially one component.

Reply:
haha, i disagree with your disagreement :-)

we will probably have to rewrite this component substantially a few times, and as i mention, rewrites are much easier if more things derive their display state from store. this is the convention we've adopted throughout the codebase, and it's really quite nice. I'd ask that you trust me on this one and give it a shot :-)